### ISSUE TITLE
Avoid double indentation when unnecessary

### ISSUE THREAD

**Issue by ego-thales** [psf/black#4505]
Hello,

I'd like your opinion on `black`'s handling of the following case and what I think would be preferable
```python
# Original
A = func([fooooooooooooooooooooooooooooooooooooooo for bar in baaaaaaaaaaaaaaaaaaaaaaaaaaaaaz])

# Current black
A = func(
    [
        fooooooooooooooooooooooooooooooooooooooo
        for bar in baaaaaaaaaaaaaaaaaaaaaaaaaaaaaz
    ]
)

# Preferred
A = func([
    fooooooooooooooooooooooooooooooooooooooo
    for bar in baaaaaaaaaaaaaaaaaaaaaaaaaaaaaz
])
```

Do you think it would be reasonable to hope for in future developments of `black`?

Thanks for your time.

**Comment by JelleZijlstra** [comment]
This is already in the unstable style, as documented at https://black.readthedocs.io/en/stable/the_black_code_style/future_style.html#labels-hug-parens . However, we can't stabilize this until a number of issues and crashes have been fixed (#4036, #4098, #4099).

**Comment by ego-thales** [comment]
Very promising! Thank you for your work and your answer :)
